src/pages/DecipherPage/Redirection.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from 'react'
+import { Redirect } from 'react-router-dom'
+
+
+
+const Redirection = ({ role }) => {
+
+   const redirectRule = () => {
+
+        if(role.isAdmin) {
+            return <Redirect to="/admin/dashboard" />
+        }
+
+        if(role.isUser) {
+            return <Redirect to="/dashboard" />
+        }
+
+        if(role.isTrader) {
+            return <Redirect to="/trader/dashboard" />
+        }
+
+    }
```

Comment:
I feel making this a component that takes in a prop `role` makes it better, not like your implementation breaks anything by the way.

Reply:
```
import React from 'react'
import { Redirect } from 'react-router-dom'

const RedirectRule = ({role}) => {

    if(role.isAdmin) {
        return <Redirect to="/admin/dashboard" />
    }

    if(role.isUser) {
        return <Redirect to="/dashboard" />
    }

    if(role.isTrader) {
        return <Redirect to="/trader/dashboard" />
    }

}

const Redirection = ({ role }) => <RedirectRule role={role} />


export default Redirection
```

Reply:
Or better still no need for the `Redirection` component

Reply:
I don't get this... This is a component that takes a prop `role`

Reply:
Also, it just occured to me that it'll be better to have a loading message for when `isLoading` is true then when we have a UI we can make it better, for now it can be `Loading...`

Reply:
We will create another card for this...

Reply:
Alright, that's better

Reply:
> I don't get this... This is a component that takes a prop `role`

So inside the `Redirection` component you created another function and called it, I don't think there's any need for that extra function.

Reply:
Hmmmm

Reply:
https://github.com/2kama/tpa/pull/7 

Re-did this here...take a look